src/services/redisSessionStorage.ts

```diff
@@ -125,26 +149,79 @@ export default class RedisSessionStorage implements ISessionStorage {
       );
     });
 
-    const [
-      deleteSessionTokenResult,
-      deleteWalletTokenResult
-    ] = await Promise.all([deleteSessionToken, deleteWalletToken]);
+    const deletePromises = await Promise.all([
+      ...deleteSessionTokens,
+      deleteWalletToken
+    ]);
 
-    if (isLeft(deleteSessionTokenResult) || isLeft(deleteWalletTokenResult)) {
+    if (
+      deletePromises.some(_ => {
+        return isLeft(_) || !_.value;
+      })
+    ) {
       return left<Error, boolean>(new Error("Error deleting the token"));
     }
+    return right<Error, boolean>(true);
+  }
 
-    if (!deleteSessionTokenResult.value || !deleteWalletTokenResult.value) {
-      return left<Error, boolean>(new Error("Error deleting the token"));
+  public async listUserSessions(
+    user: User
+  ): Promise<Either<Error, SessionsList>> {
+    const keys = await new Promise<Either<Error, ReadonlyArray<string>>>(
+      resolve => {
+        this.redisClient.keys(
```

Comment:
again, this is very inefficient, we must use a `Set`

Reply:
fixed in b90f2d6